test/e2e/framework/k8s.go

```diff
@@ -143,16 +143,21 @@ func (f *Framework) EnsureDeployment(deployment *appsv1.Deployment) *appsv1.Depl
 }
 
 // waitForPodsReady waits for a given amount of time until a group of Pods is running in the given namespace.
-func waitForPodsReady(kubeClientSet kubernetes.Interface, timeout time.Duration, expectedReplicas int, namespace string, opts metav1.ListOptions) error {
+func waitForPodsReady(kubeClientSet kubernetes.Interface, timeout time.Duration, expectedReplicas int, namespace string, opts *metav1.ListOptions) error {
+	//nolint:staticcheck // TODO: will replace it since wait.PollImmediate is deprecated
```

Comment:
Create a followup issue to track: https://github.com/kubernetes/ingress-nginx/issues/10305